kamon-core/src/main/scala/kamon/trace/Sampler.scala

```diff
@@ -44,8 +47,10 @@ object Sampler {
     val upperBoundary = Long.MaxValue * chance
     val lowerBoundary = -upperBoundary
 
-    override def decide(spanID: Long): Boolean =
-      spanID >= lowerBoundary && spanID <= upperBoundary
+    override def decide(operationName: String, builderTags: Map[String, String]): SamplingDecision = {
```

Comment:
To be honest, it's just this internal thing telling me that comparing/generating Longs is cheaper than doing it with Doubles :/.. no better justification than that.